Takeoff Mode implementation

OABF opened the pull request:
Created new classes for TakeoffMode and its stages and removed corresponding functions from pathStateClass
Created Unit test file for TakeoffMode

OABF committed:
Takeoff Mode implemented
Created new classes for TakeoffMode and its stages and removed
corresponding dunctions from pathStateClassOB
Created Unit test file

Pop0097 commented on Autopilot/CMakeLists.txt:
```diff
@@ -120,6 +120,7 @@ elseif(${KIND_OF_BUILD} STREQUAL "UNIT_TESTS")
     ${CMAKE_CURRENT_SOURCE_DIR}/PathManager/Inc/cruisingMode
     ${CMAKE_CURRENT_SOURCE_DIR}/PathManager/Inc/landingMode
+    ${CMAKE_CURRENT_SOURCE_DIR}/PathManager/Inc/takeoffMode
     ${CMAKE_CURRENT_SOURCE_DIR}/TelemetryManager/Inc
```
Run `./Tools/build.bash -c`. It will fail because you forgot to add: `PathManager/Inc/takeoffMode` in the `include_directories` on line 26

Pop0097 committed:
Fixed unit tests

Pop0097 commented on Autopilot/PathManager/Src/takeoffMode/takeoffMode.cpp:
```diff
@@ -0,0 +1,28 @@
+    return singleton;
+}
+
+void TakeoffMode::execute(Telemetry_PIGO_t telemetry_in, SFOutput_t sensor_fusion_in, IMU_Data_t imu_data_in) {
```
Make sure brackets follow style guide: https://uwarg-docs.atlassian.net/wiki/spaces/ZP/pages/1827635203/Style+Guide

Pop0097 commented on Autopilot/PathManager/Src/takeoffMode/takeoffModeStages.cpp:
```diff
@@ -0,0 +1,182 @@
+    #endif
+    _input.sensorOutput = &sf_data;
+
+    if(_input.sensorOutput->altitude > (TakeoffRollStage::_takeoff_point.altitude + EXIT_TAKEOFF_ALTITUDE)) {
```
`if (`

https://uwarg-docs.atlassian.net/wiki/spaces/ZP/pages/1827635203/Style+Guide

Pop0097 commented on Autopilot/PathManager/Src/takeoffMode/takeoffModeStages.cpp:
```diff
@@ -0,0 +1,182 @@
+    }
+
+    //if the airspeed is greater than the current desired rotation speed, set stage to climb
+    if(_input.sensorOutput->airspeed > (LandingTakeoffManager::desiredRotationSpeed(takeoff_mode->getIsPackage()))) {
+        takeoff_mode->setCurrentStage(&TakeoffClimbStage::getInstance());
+        takeoff_mode->setCurrentStageEnum(TAKEOFF_CLIMB);
+    }
+
+    if(TakeoffRollStage::_waypoint_status != WAYPOINT_SUCCESS) {
+        takeoff_mode->getModeSelector()->setIsError(true);
+    }
```
`if (`

https://uwarg-docs.atlassian.net/wiki/spaces/ZP/pages/1827635203/Style+Guide

OABF committed:
Fixed unit test error and styling mistake
Added takeoffMode.cpp and takeoffModeStages.cpp to CMake file
Fixed brackets according to the style guide requirements

Pop0097 commented on Autopilot/PathManager/Inc/takeoffMode/takeoffMode.hpp:
```diff
@@ -0,0 +1,97 @@
+        bool is_package;
+};
+
+#endif
```
newline error

OABF committed:
Fixed newline error
fixed newline error in takeoffMode.hpp
